libsolutil/CommonData.h

```diff
@@ -267,17 +267,21 @@ template<
 	typename MapType,
 	typename KeyType,
 	typename ValueType = std::decay_t<decltype(std::declval<MapType>().find(std::declval<KeyType>())->second)> const&,
-	typename AllowCopyType = void*
+	typename AllowCopyType = std::conditional_t<std::is_pod_v<ValueType> || std::is_pointer_v<ValueType>, detail::allow_copy, void*>
 >
-decltype(auto) valueOrDefault(MapType&& _map, KeyType const& _key, ValueType&& _defaultValue = {}, AllowCopyType = nullptr)
+decltype(auto) valueOrDefault(
+	MapType&& _map,
+	KeyType const& _key,
+	ValueType&& _defaultValue = {},
+	AllowCopyType = {})
```

Comment:
Just curious: What does an empty initializer list as a default for a type mean?

Reply:
That's not the type, but the argument ``AllowCopyType _allowCopy = {}`` just without mentioning the name :-).

Reply:
Ah I see! It may be good to name all parameters just for consistency :-)

Reply:
It would trigger an "argument is unused" error, though :-). It's this weird meta-programming trick, in which just the type of the argument is really relevant, not its value - and with the our warning-as-errors settings I can't still give it a name :-).

Reply:
Oh I see! Please ignore my comment in that case :-)